frontend/src/metabase/entities/search.js

```diff
@@ -22,20 +22,30 @@ export default createEntity({
   api: {
     list: async (query = {}) => {
       if (query.collection) {
-        const { collection, archived, model, ...unsupported } = query;
+        const {
+          collection,
+          archived,
+          model,
+          writeable,
+          ...unsupported
+        } = query;
         if (Object.keys(unsupported).length > 0) {
           throw new Error(
             "search with `collection` filter does not support these filters: " +
               Object.keys(unsupported).join(", "),
           );
         }
-        return (await collectionList({ collection, archived, model })).map(
-          item => ({
-            collection_id: canonicalCollectionId(collection),
-            archived: archived || false,
-            ...item,
-          }),
-        );
+        return (await collectionList({
+          collection,
+          archived,
+          model,
+          writeable,
+        })).map(item => ({
+          collection_id: canonicalCollectionId(collection),
+          archived: archived || false,
+          writeable: writeable || false,
```

Comment:
This makes it so we're always passing writable as either true or false. Maybe this makes sense as an alternate API?
1. `writeable: true` – writable collections
2. `writeable: false` – unwritable collections
3.  [not passed] – all collections

I'm guessing `writeable: false` is currently retrieving all collections. That works if we're sure we'll never want to filter down just unwritable collections.

Reply:
I think I can actually take this line out, it's trying to add a `writeable` key to the items returned from the API call (I think?) but we don't use that so I don't think it's needed

Reply:
Ah you're right!